presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java

```diff
@@ -125,4 +126,46 @@ public void testJsonFileBasedFunction()
         assertQuery("SELECT json.test_schema.map(json.test_schema.array_constructor(linenumber), json.test_schema.array_constructor(linenumber)) FROM lineitem", "SELECT map(array_constructor(linenumber), array_constructor(linenumber)) FROM lineitem");
         assertQuery("SELECT json.test_schema.map_entries(json.test_schema.map(json.test_schema.array_constructor(linenumber), json.test_schema.array_constructor(linenumber))) FROM lineitem", "SELECT map_entries(map(array_constructor(linenumber), array_constructor(linenumber))) FROM lineitem");
     }
+
+    /**
+     * Test aggregation using companion functions with partial and final aggregation steps handled by separate queries.
+     * The first query computes partial aggregation states and stores them in the avg_partial_states table.
+     * Subsequent queries read from avg_partial_states and aggregate the states to the final result.
+     */
+    @Test
+    public void testAggregationCompanionFunction()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty("table_writer_merge_operator_enabled", "false")
+                .setCatalogSessionProperty("hive", "collect_column_statistics_on_write", "false")
+                .setCatalogSessionProperty("hive", "optimized_partition_update_serialization_enabled", "false")
+                .build();
+        try {
+            getQueryRunner().execute(session,
+                    "CREATE TABLE avg_partial_states AS ( "
+                            + "SELECT orderpriority, cast(json.test_schema.avg_partial(shippriority) as ROW(sum DOUBLE, count BIGINT)) as states "
+                            + "FROM orders "
+                            + "GROUP BY orderstatus, orderpriority "
+                            + ")");
+
```

Comment:
nit: `CREATE TABLE` is better off outside of try-finally since it's failure does not need the table to be dropped; this follows the principle of minimizing try-blocks' scope.

Reply:
Hi @Ali-P, I saw that test cases in AbstractTestNativeGeneralQueries.java put CREATE TABLE inside the try-blocks, so I followed that. I don't have a strong opinion. Let me know what you think.

Reply:
I see; leaving it outside of try-catch is better (one would need a reason for including a statement in the try-block, which there is none here). But the PR is GTG (as the PR is similar to other examples). No need to update. :)